docs/src/main/sphinx/optimizer/pushdown.rst

```diff
@@ -151,6 +151,65 @@ to perform the remaining query processing on a smaller amount of data.
 The specifics for the supported pushdown of table joins varies for each data
 source, and therefore for each connector.
 
+However, there are some generic conditions that must be met in order for a join
+to be pushed down:
+
+* join pushdown must be enabled
+* all predicates that are part of the join must be possible to be pushed down
+* none of the tables in the join must be empty
+* all tables in the join must be from the same catalog
+
+You can check if pushdown for a specific join is performed by looking at the
+:doc:`EXPLAIN plan </sql/explain>` of the query. If a join is successfully
+pushed down to the connector, the explain plan does **not** show that ``Join`` operator.
+The explain plan only shows the operations that are performed by Trino.
+
+As an example, we loaded the TPCH data set into a PostgreSQL database and then
```

Comment:
I think that the documentation should tell the user how to setup himself a test postgres instance used to reproduce the pushdown join behavior.
I'd rather start from the supposition that the postgres database table is set and it contains two tables.

Reply:
How to setup the TPCH  connector [is documented](https://trino.io/docs/current/connector/tpch.html) already. Configuring a Postgres connector is [also documented](https://trino.io/docs/current/connector/postgresql.html). Once you have those connectors configured, its just a `CREATE TABLE AS SELECT` statement to load TPCH date into a PostgreSQL database.

Reply:
Sorry. I've meant to say that IMO the join push down should concentrate on what join push down is and not how to set up a postgres database in order to test it.